lib/service/service.go

```diff
@@ -3399,6 +3399,13 @@ func (process *TeleportProcess) initProxyEndpoint(conn *Connector) error {
 		} else {
 			log.Infof("Shutting down gracefully.")
 			ctx := payloadContext(payload, log)
+			if tsrv != nil {
+				if drainer, ok := tsrv.(interface {
+					Drain(context.Context) error
+				}); ok {
```

Comment:
Can't we just add `Drain` (or maybe `PreShutdown`) to `lib/reversetunnel.Server` instead of doing this check?

Reply:
Added here https://github.com/gravitational/teleport/pull/12661/commits/6f477d09b351435cd77ae5ae827ad96d7182c7f6. I was expecting other places to mock or implement the interface but turns out it was an easy change.